### DISCUSSION TITLE
SNOW vs SNOW2 Peak Detection

### DISCUSSION THREAD

**Discussion by Jack-Waghorne** [PMEAL/porespy#938] [Q&A] [answered]
Hi PoreSpy Team,

I am interested in understanding the distinctions between SNOW2 and the original SNOW algorithm, specifically around peak detection and trimming. As the documentation on your website suggests, there are alterations in the saddle peak trimming to make it less aggressive, although the reference it cites is denoted as 'PRE' without a hyperlink. I have looked through the SNOW2 documentation here and on the website and cannot find where these changes are described. Could you please provide some guidance on this? 

Website ref: https://porespy.org/examples/filters/reference/snow_partitioning.html

Thanks in advance,
Jack

P.S. I think the toolbox is great and has greatly aided me in understanding porous media modelling, thanks for outputting and supporting the community!

**Comment by jgostick** [comment, marked as answer]
Hi @Jack-Waghorne.  Snow "1" removed too many peaks because of a bug that showed up if the image resolution was too poor.  Basically it was overwriting some nearby peaks when inserting new peaks.  We have added the ability to specify your own peaks in snow2 if you really want.

**Reply by Jack-Waghorne** [reply]
Hi @jgostick,

Thank you for your swift support with this. I can see how this would be an issue. So does the V2 incorporate new conditions to end the trimming algorithm, ie interpreting neighbouring peaks and their relationship to the peak being assessed?

A quick aside query, something that struck me as strange was that peaks often shown in examples are present in clusters. Is there a rationale as to why these clusters are not thinned or condensed into a single point? And when it comes to the final trim of removing neighbouring peaks, how are the coordinates of each peak determined, is it the centroid of the clusters and the comparative distance to solid space the distance map value at that centroid? Some clarity on this would be greatly appreciated. 

Thanks in advance,
Jack
